prov/verbs/src/fi_verbs.h

```diff
@@ -984,50 +986,61 @@ vrb_send_iov_flags(struct vrb_ep *ep, struct ibv_send_wr *wr,
 		      uint64_t flags)
 {
 	size_t len = 0;
-	enum fi_hmem_iface iface;
-	uint64_t device;
+	enum fi_hmem_iface iface = FI_HMEM_SYSTEM;
+	uint64_t device = 0;
 	void *bounce_buf;
-	void *send_desc;
+	void *send_desc = NULL;
 	ssize_t ret;
 
-	if (!desc) {
-		vrb_set_sge_iov_inline(wr->sg_list, iov, count, len);
-		iface = FI_HMEM_SYSTEM;
-		device = 0;
-		send_desc = NULL;
-	} else {
-		vrb_set_sge_iov_count_len(wr->sg_list, iov, count, desc, len);
-		iface = ((struct vrb_mem_desc *) desc[0])->iface;
-		device = ((struct vrb_mem_desc *) desc[0])->device;
-		send_desc = desc[0];
-	}
+	if (ep->hmem_enabled) {
+		if (!desc) {
+			vrb_set_sge_iov_inline(wr->sg_list, iov, count, len);
+		} else {
+			vrb_set_sge_iov_count_len(wr->sg_list, iov, count, desc,
+						  len);
+			iface = ((struct vrb_mem_desc *) desc[0])->iface;
+			device = ((struct vrb_mem_desc *) desc[0])->device;
+			send_desc = desc[0];
+		}
```

Comment:
This is removing a change that was just added in a previous patch.  There needs to be a cleaner way of making these changes.  Otherwise trying to cherry pick bug fixes that follow become a mess.

Reply:
Will fix.